cluster.go

```diff
@@ -459,13 +474,73 @@ func (c *Cluster) Do(a Action) error {
 	return c.doInner(a, addr, key, false, doAttempts)
 }
 
+func (c *Cluster) getClusterDownSince() int64 {
+	return atomic.LoadInt64(&c.lastClusterdown)
+}
+
+func (c *Cluster) setClusterDown(down bool) (changed bool) {
+	prevVal := atomic.LoadInt64(&c.lastClusterdown)
```

Comment:
So initially I was going to ask if we could just do an atomic swap here instead of a load and CAS, as the code would be a lot less complex, but then I did a benchmark and swap is ~10x less performant than load. Can you put a comment to that effect here, in case anyone comes along later wondering why this section is so much more complex than necessary?

Reply:
After a bit of sleep, reading your comments and running a few benchmarks, I realized that we can remove the CAS-loop and replace it with a swap without losing on performance, but winning on simplicity and readability.

Also added two comments to explain the logic a bit better.